File: src/0965.univalued-binary-tree.1005/solution.go
package univaluedbinarytree1005

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func isUnivalTree(root *TreeNode) bool {
	if root == nil {
		return true
	}

	return fn(root.Left, root.Val) && fn(root.Right, root.Val)
}

func fn(node *TreeNode, target int) bool {
	if node == nil {
		return true
	}
	return node.Val == target && fn(node.Left, target) && fn(node.Right, target)
}

File: src/0965.univalued-binary-tree.1005/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 * int val;
 * TreeNode left;
 * TreeNode right;
 * TreeNode() {}
 * TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) {
 * this.val = val;
 * this.left = left;
 * this.right = right;
 * }
 * }
 */
class Solution {
    public boolean isUnivalTree(TreeNode root) {
        if (root == null) {
            return true;
        }
        return isSameValue(root, root.val);
    }

    private boolean isSameValue(TreeNode root, int val) {
        if (root == null) {
            return true;
        }
        return root.val == val && isSameValue(root.left, val) && isSameValue(root.right, val);
    }
}